### PR TITLE
ZFIN-8986: Update marker relationships when renaming construct

### PR DESCRIPTION
**Pull request by rtaylorzfin**
- Add ability to generate a ConstructName object by fetching the ConstructComponent objects from the DB.
- When renaming a construct, make sure to add to marker_relationship and construct_marker_relationship (and delete).
- Add new construct change: ZDB-TGCONSTRCT-190812-12

### CODE REVIEW

File: source/org/zfin/construct/presentation/ConstructComponentService.java [hunk 1]
```diff
@@ -308,5 +314,127 @@ private static void validateConstructNameOrThrowException(String constructName)
...
+                parsingCassette = "promoter";
+            }
+            if (currentComponent.getComponentValue().equals(COMPONENT_SEPARATOR)) {
+                continue;
+            } else if (currentComponent.getComponentValue().equals(")") && !componentIterator.hasNext()) {
+                continue;
+            } else if (currentComponent.getComponentCategory().equals("promoter component")) {
+                currentPromoter.addPromoterPart(currentComponent.getComponentValue());
+            } else if (currentComponent.getComponentCategory().equals("coding component")) {
+                parsingCassette = "coding";
```
Comment (cmpich) [inline source/org/zfin/construct/presentation/ConstructComponentService.java:368]: Can that we re-written with a switch statement?

Comment (rtaylorzfin) [reply source/org/zfin/construct/presentation/ConstructComponentService.java:368]: Good idea. I'll change it

Comment (rtaylorzfin) [reply source/org/zfin/construct/presentation/ConstructComponentService.java]: Updated

**Review by cmpich — APPROVED** [review]
Looks good.

**Pull request merged**